packages/wallets/src/features/cashier/modules/WithdrawalCrypto/components/WithdrawalCryptoForm/components/WithdrawalCryptoAmountConverter/WithdrawalCryptoAmountConverter.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Field, FieldProps, useFormikContext } from 'formik';
+import { useActiveWalletAccount, useCurrencyConfig, useExchangeRate } from '@deriv/api';
+import ArrowBold from '../../../../../../../../public/images/arrow-bold.svg';
+import { WalletTextField } from '../../../../../../../../components';
+import type { TForm } from '../../WithdrawalCryptoForm';
+import './WithdrawalCryptoAmountConverter.scss';
+
+type TProps = {
+    activeWallet: ReturnType<typeof useActiveWalletAccount>['data'];
+    exchangeRate: ReturnType<typeof useExchangeRate>['data'];
+    getCurrencyConfig: ReturnType<typeof useCurrencyConfig>['getConfig'];
+};
+
+const helperMessageMapper = {
+    decimalPlacesExceeded: (limit: number) => `Up to ${limit} decimal places are allowed.`,
+    insufficientFunds: 'Insufficient funds',
+    invalidInput: 'Should be a valid number.',
+    withdrawalLimitError: (min: string, max: string, currency: string) => {
+        return `The current allowed withdraw amount is ${min} to ${max} ${currency}.`;
+    },
+};
+
+const WithdrawalCryptoAmountConverter = ({ activeWallet, exchangeRate, getCurrencyConfig }: TProps) => {
+    const [isCryptoInputActive, setIsCryptoInputActive] = useState(false);
+    const { errors, setValues, values } = useFormikContext<TForm>();
+    const FRACTIONAL_DIGITS_CRYPTO = activeWallet?.currency
+        ? getCurrencyConfig(activeWallet?.currency)?.fractional_digits
+        : 8;
+    const FRACTIONAL_DIGITS_FIAT = getCurrencyConfig('USD')?.fractional_digits;
+    const MINIMUM_WITHDRAWAL_AMOUNT = activeWallet?.currency
+        ? getCurrencyConfig(activeWallet?.currency)?.minimum_withdrawal
+        : 0;
+
+    const validateCryptoInput = (value: string) => {
```

Comment:
Given numerous edge cases and quite a lot of complex functionality, would it make sense to write tests? For all the future devs who sooner or later will need to update the form and validation.